test/unit/features/instance/methods-lifecycle.spec.js

```diff
@@ -53,6 +53,37 @@ describe('Instance methods lifecycle', () => {
         }
       }).$mount()
     })
+
+    it('Dep.target should be undefined during invocation of child immediate watcher', done => {
+      let calls = 0, childData
+      const parentUpdate = jasmine.createSpy()
+      new Vue({
+        template: '<div><my-component></my-component></div>',
+        updated: parentUpdate,
+        components: {
+          myComponent: {
+            template: '<div>{{ a }}</div>',
+            data() {
+              return childData = { a: 123 }
```

Comment:
Let's move the data variable outside of new Vue, and return it here

```suggestion
              return childData
```